### PR TITLE
 tikv_util: support get tikv api version via store command

### LINKED ISSUE #11025: Read API version from kvdb via tikv-ctl
Show data encode version (aka. api version) by `tikv-ctl store` command. This should work remotely as well as locally. The data encode in kvdb is defined in https://github.com/tikv/tikv/pull/11021.

### PR DESCRIPTION
**Pull request by yuqi1129**
close #11025

Signed-off-by: yuqi1129 <yuqi4733@gmail.com>

### What problem does this PR solve?

Issue Number: close #11025 
Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None.
```

### CODE REVIEW

File: src/server/debug.rs [hunk 1]
```diff
@@ -813,6 +819,19 @@ impl<ER: RaftEngine> Debugger<ER> {
             })
     }
 
+    pub fn get_store_info(&self) -> Result<DebugStoreInfo> {
```
Comment (andylokandy) [inline src/server/debug.rs:822]: get_api_version maybe better since we already have the `get_cluster_id` and `get_store_id`.

Comment (yuqi1129) [reply src/server/debug.rs:822]: @andylokandy , `get_api_version ` only get api version information, When execute `store` command, we expect output valuef of store_id and api_version, if change to   `get_api_version `, I'am afraid we may call the same method and get data from rocksdb twice

Comment (andylokandy) [reply src/server/debug.rs:822]: Can we refactor all these threes into a single `get_store_indent`?

File: src/server/debug.rs [hunk 2]
```diff
@@ -809,36 +803,13 @@ impl<ER: RaftEngine> Debugger<ER> {
...
-    pub fn get_cluster_id(&self) -> Result<u64> {
-        let db = &self.engines.kv;
-        db.get_msg::<StoreIdent>(keys::STORE_IDENT_KEY)
-            .map_err(|e| box_err!(e))
-            .and_then(|ident| match ident {
-                Some(ident) => Ok(ident.get_cluster_id()),
-                None => Err(Error::NotFound("No cluster ident key".to_owned())),
+            .map(|ident| match ident {
+                Some(ident) => ident,
+                None => StoreIdent::default(),
```
Comment (andylokandy) [inline src/server/debug.rs:812]: Should we throw an error here?

File: src/server/debug.rs (hunk 1 above)
Comment (yuqi1129) [reply src/server/debug.rs]: Sounds reasonable. i have merge three into one

**Comment by yuqi1129** [comment]
Can anyone help to reivew this PR?

**Review by andylokandy — APPROVED** [review]

**Comment by andylokandy** [comment]
@zhongzc  PTAL

File: src/server/debug.rs [hunk 3]
```diff
@@ -1486,30 +1477,25 @@ mod tests {
...
         }
 
-        fn set_store_id(&self, store_id: u64) {
-            if let Ok(mut ident) = self.get_store_ident() {
-                ident.set_store_id(store_id);
-                let db = &self.engines.kv;
-                db.put_msg(keys::STORE_IDENT_KEY, &ident).unwrap();
-            }
+        fn set_cluster_id(&self, cluster_id: u64) {
+            let mut ident = self.get_store_ident().unwrap_or_default();
```
Comment (zhongzc) [inline src/server/debug.rs:1488]: `unwrap_or_default` is not compatible with the origin logic. Will it be a problem?

File: src/server/debug.rs [hunk 4]
```diff
@@ -1486,30 +1477,25 @@ mod tests {
...
-        fn get_store_ident(&self) -> Result<StoreIdent> {
+        fn set_store_id(&self, store_id: u64) {
+            let mut ident = self.get_store_ident().unwrap_or_default();
+            ident.set_store_id(store_id);
             let db = &self.engines.kv;
-            db.get_msg::<StoreIdent>(keys::STORE_IDENT_KEY)
-                .map_err(|e| box_err!(e))
-                .map(|ident| match ident {
-                    Some(ident) => ident,
-                    None => StoreIdent::default(),
```
Comment (yuqi1129) [inline src/server/debug.rs:1495]: @zhongzc  Not really. This is the original logic

File: src/server/debug.rs (hunk 3 above)
Comment (yuqi1129) [reply src/server/debug.rs:1488]: @zhongzc Not really. This is the original logic
image

Comment (zhongzc) [reply src/server/debug.rs:1488]: @yuqi1129 

original (pseudocode):
```
match db.get_msg:
    case Ok(Some(id)) => id
    case Ok(None) => StoreIdent::default()
    case Err(_) => noop
```

here (pseudocode):
```
match db.get_msg:
    case Ok(Some(id)) => id
    case Ok(None) => StoreIdent::default()
    case Err(_) => StoreIdent::default()
```

Comment (zhongzc) [reply src/server/debug.rs:1488]: Well, I just found out that it's just test code. Not a big problem.

**Review by zhongzc — APPROVED** [review]

**Pull request merged**